lighthouse-core/gather/gatherers/accessibility.js

```diff
@@ -57,43 +57,53 @@ async function runA11yChecks() {
   // are relative to the top of the page
   document.documentElement.scrollTop = 0;
 
-  /** @param {import('axe-core/axe').Result} result */
-  const augmentAxeNodes = result => {
-    result.nodes.forEach(node => {
-      // @ts-expect-error - getNodeDetails put into scope via stringification
-      node.node = getNodeDetails(node.element);
-      // @ts-expect-error - avoid circular JSON concerns
-      node.element = node.any = node.all = node.none = undefined;
-    });
-
-    // Ensure errors can be serialized over the protocol
-    /** @type {(Error & {message: string, errorNode: any}) | undefined} */
-    // @ts-expect-error - when rules error axe sets these properties
-    // see https://github.com/dequelabs/axe-core/blob/eeff122c2de11dd690fbad0e50ba2fdb244b50e8/lib/core/base/audit.js#L684-L693
-    const error = result.error;
-    if (error instanceof Error) {
-      // @ts-expect-error
-      result.error = {
-        name: error.name,
-        message: error.message,
-        stack: error.stack,
-        errorNode: error.errorNode,
-      };
-    }
+  return {
+    violations: axeResults.violations.map(createAxeRuleResultArtifact),
+    incomplete: axeResults.incomplete.map(createAxeRuleResultArtifact),
+    notApplicable: axeResults.inapplicable.map(result => ({id: result.id})),
+    version: axeResults.testEngine.version,
   };
+}
+
+/**
+ * @param {import('axe-core/axe').Result} result
+ * @return {LH.Artifacts.AxeRuleResult}
+ */
+function createAxeRuleResultArtifact(result) {
+  // Simplify `nodes` and collect nodeDetails for each.
+  const nodes = result.nodes.map(node => {
+    const {html, target, failureSummary, element} = node;
+    // TODO: with `elementRef: true`, `element` _should_ always be defined, but need to verify.
```

Comment:
very debatable TODO. [The type](https://github.com/dequelabs/axe-core/blob/b297571cf639a61c519e84e25fd268aab11cfb07/axe.d.ts#L126) is `{element?: HTMLElement}`, true, and we could be defensive, but AFAICT with `elementRef: true` `element` is always set.

- On the one hand I haven't been able to verify that it's always set (there doesn't seem to be a single place in the code that sets it, but maybe I'm missing it).
- On the other hand we've been assuming that it's always set since 2016.

So...worth marking it TODO or just a comment?

Reply:
Ideally we could generate a warning if this invariant fails (and when it does, just skip this node and not show it in the results / or keep and fake a nodeDetails / use the body node).  would require a not-ideal restructure of these two functions, so maybe not worth it generating a warning.

am hesitant to have it fail the entire run, but it is an option to just throw an error if element does not exist..

Reply:
> Ideally we could generate a warning if this invariant fails (and when it does, just skip this node and not show it in the results / or keep and fake a nodeDetails / use the body node). would require a not-ideal restructure of these two functions, so maybe not worth it generating a warning.

what do you think of beaconing to Sentry? I assume we'd get some kind of `Cannot read property 'whatever' of undefined` if we really did sometimes have no element, but I can't find any hint of that in Sentry for the accessibility gatherer or audits. But this way it would be explicit and we'd have as good a signal as any after a month that it really can't ever happen?